### PR TITLE
Reduce allocations by 2 for each createQuery call.

### PR DESCRIPTION
**Pull request by JakeWharton**
* Use single Action0 instance for every stream to check whether it is being subscribed to in a transaction.
* Promote anonymous Query subclass to named inner-class. This allows it to implement Func1 which returns itself for use as a mapper from triggered tables to the query instance.

### CODE REVIEW

File: sqlbrite/src/main/java/com/squareup/sqlbrite/BriteDatabase.java [hunk 1]
```diff
@@ -283,55 +290,22 @@ public QueryObservable createQuery(@NonNull final Iterable<String> tables, @NonN
   }
 
   @CheckResult @NonNull
-  private QueryObservable createQuery(final Func1<Set<String>, Boolean> tableFilter,
-      final String sql, final String... args) {
+  private QueryObservable createQuery(Func1<Set<String>, Boolean> tableFilter, String sql,
```
Comment (AlecStrong) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/BriteDatabase.java]: whats the reason for changing these to non-final?

Comment (JakeWharton) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/BriteDatabase.java]: They were only `final` for use inside the anonymous `Query` class previously declared inside this method. Now they're just handed off to the `DatabaseQuery` constructor so the `final` modifier had no purpose.

Comment (AlecStrong) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/BriteDatabase.java]: sounds good. knowing you i figured it was eliminating some synthetic accessor method or something

**Comment by AlecStrong** [comment]
lgtm!

**Comment by JakeWharton** [comment]
This should also reduce the method count by 1! 🎉 (The constructor of the previously-separate `Func1` class)

**Pull request merged**